spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample/spring-cloud-gcp-integration-pubsub-sample-sender/src/main/java/com/example/SenderApplication.java

```diff
@@ -66,6 +66,6 @@ public void onSuccess(String result) {
 	@MessagingGateway(defaultRequestChannel = "pubSubOutputChannel")
 	public interface PubSubOutboundGateway {
 
-		void sendToPubSub(String text);
+		void sendToPubSub(byte[] text);
```

Comment:
Can't we really make an explicit `String -> byte[]` conversion in the `PubSubMessageHandler`?

Seems for me not a big deal to call there `getBytes()` instead of requesting some external converters/paylaodMappers.

Reply:
That would make us also ask users for an encoding at the adapter level, which is totally fine by me.
But that would be better suited for another PR I think
@meltsufin wdyt?

Reply:
Correct. So, can we hold merging this PR off until we come up with the solution for target adapters?

Reply:
This PR fixes the sample to be according to the current state of `master`, which should have been done on #731.

If anything, I think this PR should be merged and, later, if we decide on any such changes, they should be accompanied by the respective sample changes.

Reply:
OK! So, the next time let's consider to fix samples together with the actual core changes.
Otherwise it doesn't looks like productive development: we are going to merge the change we we already know will be dropped by the next change. 🤷‍♂️

Reply:
In the future, we will have the samples run automatically as part of build and will catch these kinds of issues. See: #705 .